api/src/main/java/org/eclipse/microprofile/opentracing/ClientTracingRegistrar.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.opentracing;
+
+import java.util.ServiceLoader;
+import java.util.concurrent.ExecutorService;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+
+/**
+ * This class registers tracing components into {@link ClientBuilder}.
+ * Implementation might enable client tracing globally, then this explicit configuration has no effects.
+ *
+ * To enable explicit configuration call {@link ClientTracingRegistrar#configure(ClientBuilder)} which
+ * return a client with enabled tracing integration.
+ *
+ * @author Pavol Loffay
+ */
+public class ClientTracingRegistrar {
+
+    private ClientTracingRegistrar() {}
+
+    /**
+     *  Register tracing components into client builder instance.
+     *
+     * @param clientBuilder client builder
+     * @return client with tracing integration
+     */
+    public static Client configure(ClientBuilder clientBuilder) {
+        for(ClientTracingRegistrarProvider registrar: ServiceLoader.load(ClientTracingRegistrarProvider.class)) {
+            return registrar.configure(clientBuilder);
+        }
+        return clientBuilder.build();
+    }
+
+    /**
+     *  Register tracing components into client builder instance.
+     *
+     * @param clientBuilder client builder
+     * @param executorService executorService which will be added to the client. Note that this overrides
+     *      executor service added previously to the client.
+     * @return client with tracing integration
+     */
+    public static Client configure(ClientBuilder clientBuilder, ExecutorService executorService) {
+        for(ClientTracingRegistrarProvider registrar: ServiceLoader.load(ClientTracingRegistrarProvider.class)) {
+            return registrar.configure(clientBuilder, executorService);
+        }
+        return clientBuilder.build();
```

Comment:
Should this return `ClientBuilder` instead?

Or do we assume the developer will always do this last?

Reply:
Returing the client (not builder) assures that executor service won't be overriden by subsequent callls on the builder. I agree that more flexible is to return the builder. Perhaps a good documentation of this issue is enough to return the builder.

Note that executor service is also overridden if it has been configured before calling `ClientTracingRegistrar.configure(builder)`, therefore there is `ClientTracingRegistrar.configure(builder, execService)` - `execService` is wrapped into ot-aware one.